dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/Request.java

```diff
@@ -82,6 +82,12 @@
         Class<? extends ListParamEncoder<?>> encoder() default DefaultListParamEncoder.class;
     }
 
+    @Retention(RetentionPolicy.SOURCE)
+    @Target(ElementType.PARAMETER)
+    @interface HeaderMap {
+        Class<? extends MultimapParamEncoder<?>> encoder() default DefaultMultimapParamEncoder.class;
+    }
```

Comment:
Requires documentation in the readme, similar to what we have for `@Request.QueryMap`